tests/template/extension_test.php

```diff
@@ -253,8 +259,197 @@ public function data_template_extensions()
+				<span class="sr-only">Pencil icon</span>',
+			],
+			/** PNG: default */
+			[
+				[
+					'type'			=> 'png',
+					'icon'			=> 'phone',
+					'title'			=> 'ICON_PHONE',
+					'hidden'		=> false,
+					'classes'		=> '',
+					'attributes'	=> [],
+				],
+				[
+					'ICON_PHONE'	=> 'Phone icon',
+				],
+				'<img class="o-icon png png-phone" src="phpBB/styles/chameleon/theme/icons/png/phone.png" alt="Phone icon" />',
+			],
+			/** PNG: all options */
+			[
+				[
+					'type'			=> 'png',
+					'icon'			=> 'pencil',
+					'title'			=> 'ICON_PENCIL',
+					'hidden'		=> true,
+					'classes'		=> 'my-class',
+					'attributes'	=> [
+						'data-url'		=> 'my-test-url/test-page.php?u=2',
+					],
+				],
+				[
+					'ICON_PENCIL'	=> 'Pencil icon',
+				],
+				'<img class="o-icon png png-pencil my-class" src="phpBB/styles/chameleon/theme/icons/png/pencil.png" alt="Pencil icon" data-url="my-test-url/test-page.php?u=2" />',
+			],
+			/** SVG: default */
+			[
+				[
+					'type'			=> 'svg',
+					'icon'			=> 'phone',
+					'title'			=> 'ICON_PHONE',
+					'hidden'		=> false,
+					'classes'		=> '',
+					'attributes'	=> [],
+				],
+				[
+					'ICON_PHONE'	=> 'Phone icon',
+				],
+				'<svg class="o-icon svg svg-phone" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" aria-labelledby="icon_phone-123456789" role="img">
+					<title id="icon_phone-123456789">Phone icon</title>
+					<path fill="none" d="M0 0h24v24H0z"></path>
+					<path d="M20.01 15.38c-1.23 0-2.42-.2-3.53-.56-.35-.12-.74-.03-1.01.24l-1.57 1.97c-2.83-1.35-5.48-3.9-6.89-6.83l1.95-1.66c.27-.28.35-.67.24-1.02-.37-1.11-.56-2.3-.56-3.53 0-.54-.45-.99-.99-.99H4.19C3.65 3 3 3.24 3 3.99 3 13.28 10.73 21 20.01 21c.71 0 .99-.63.99-1.18v-3.45c0-.54-.45-.99-.99-.99z"></path>
+				</svg>',
+			],
+			/** SVG: all options */
+			[
+				[
+					'type'			=> 'svg',
+					'icon'			=> 'pencil',
+					'title'			=> 'ICON_PENCIL',
+					'hidden'		=> true,
+					'classes'		=> 'my-svg-class',
+					'attributes'	=> [
+						'data-ajax'		=> 'my_ajax_callback',
+					],
+				],
+				[
+					'ICON_PENCIL'	=> 'Pencil icon',
+				],
+				'<svg class="o-icon svg svg-pencil my-svg-class" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" aria-hidden="true" aria-labelledby="icon_pencil-123456789" role="img" data-ajax="my_ajax_callback">
+					<title id="icon_pencil-123456789">Pencil icon</title>
+					<path d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"></path>
+					<path d="M0 0h24v24H0z" fill="none"></path>
+				</svg>',
+			],
+			/** SVG: Sanitization */
+			[
+				[
+					'type'			=> 'svg',
+					'icon'			=> 'dirty',
+					'title'			=> '',
+					'hidden'		=> false,
+					'classes'		=> '',
+					'attributes'	=> [],
+				],
+				[],
+				'<svg class="o-icon svg svg-dirty" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" role="img">
```

Comment:
what is svg-dirty?

Reply:
Most of the provided SVG files provided are 'clean' already. As in, don't contain any unnecessary or unwanted tags, such as `DOCTYPE`, `<?xml ?>`, `<title>`, etc..
The 'dirty' one does, so it test if the svg gets cleaned properly.
And if you mean the `svg-dirty` class, that's the svg file(/icon) name that gets added to to `svg-`, like with all the others, such as `svg-pencil` and `svg-phone`.

Reply:
o your adding a class for svgs? gotcha guess it does not hurt...

Reply:
I mean, if it is not necessary I will remove it. You're the master 😛 
I thought it was desired, to perhaps style specific svg's.
Custom classes are also added, as per the test above.

Reply:
it does not hurt anything, I would not recommend using it to style. Its better to style based on where something is used not what it is usually which custom classes handle